subprojects/test-runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/RemoteDeviceImpl.kt

```diff
@@ -15,26 +15,26 @@ internal class RemoteDeviceImpl(
 
     override fun disconnect(): Result<String> = processRunner.run(command = "$adb disconnect $serial")
 
-    override fun connect(): Result<String> {
+    private suspend fun connect(): Result<String> {
         disconnect()
```

Comment:
It'd great to make this function reentrant. 
If a device is already connected, we can skip the whole logic and don't reconnect.

At least, we can express a current behavior explicitly by renaming function `connect()` to `reconnect()`.

Reply:
The state of connection means nothing for us. We could think that we are already connected but ADB could disconnected without us

Reply:
Ok, sounds reasonable